src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+      if (direction === "start") {
+        this.#draggedDistanceXForResize -= currentDraggedDistanceX;
+
+        setProperty(this.el, DIALOG_DRAGGED_X, this.#draggedDistanceXForResize);
+      }
+
+      setProperty(this.el, DIALOG_INLINE_SIZE, newRestrictedInlineSize);
+    }
+  } as const;
+
+  #resizeEdgesAndCornersDictionary: {
+    [key in ChDialogResizeElement]: (dialogRect: DOMRect) => void;
+  } = {
+    "block-start": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "start"),
+
+    "block-end": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "end"),
+
+    "inline-start": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start"),
+
+    "inline-end": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end"),
+
+    "block-start-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-start-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    },
+
+    "block-end-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-end-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    }
+  };
+
+  // Refs
+  #resizeLayer: HTMLDivElement;
+  #dialogRef: HTMLDialogElement;
+
+  @Element() el: HTMLChDialogElement;
+
+  @State() dragging = false;
+  @State() relativeDialog = false;
+  @State() resizing = false;
+
+  /**
+   * Specifies the drag behavior of the dialog.
+   * If `allowDrag === "header"`, a slot with the `"header"` name will be
+   * available to place the header content.
+   */
+  @Prop() readonly allowDrag: "box" | "header" | "no" = "box";
+
+  /**
+   * Specifies whether the dialog is hidden or visible.
+   */
+  // eslint-disable-next-line @stencil-community/ban-default-true
+  @Prop({ mutable: true, reflect: true }) hidden = false;
+  @Watch("hidden")
+  handleHiddenChange(hidden: boolean) {
+    // Schedule update for watchers
+    this.#checkBorderSizeWatcher = true;
+    this.#checkPositionWatcher = true;
+
+    // Update the dialog visualization
+    if (hidden) {
+      this.#dialogRef.close();
+    } else {
+      this.#showMethod();
+    }
+  }
+
+  /**
+   * Specifies whether the dialog is a modal or not.
+   */
+  // eslint-disable-next-line @stencil-community/ban-default-true
+  @Prop() readonly modal = true;
+  @Watch("modal")
+  watchModalHandler() {
+    // Prevent DOMException "... The dialog is already open ..."
+    !this.hidden && (this.hidden = true);
+  }
```

Comment:
Why we can't change the modal property at runtime?

Also, don't use logical assignment that way, this is more readable:

```ts
if (!this.hidden) {
  this.hidden = true;
}
```

Also, that equals to:
```ts
this.hidden = true
```